Dockerfile

```diff
@@ -36,18 +42,18 @@ RUN wget https://github.com/protocolbuffers/protobuf/releases/download/v"${PROTO
 ENV GOROOT /usr/local/go
 ENV GOPATH /go
 ENV PATH $GOPATH/bin:$GOROOT/bin:$PATH
-ENV GORELEASE go1.17.linux-amd64.tar.gz
-RUN wget -q https://dl.google.com/go/$GORELEASE \
-    && tar -C $(dirname $GOROOT) -xzf $GORELEASE \
-    && rm $GORELEASE \
-    && mkdir -p $GOPATH/{src,bin,pkg}
+RUN export GORELEASE=go1.18.5.linux-$([ $(uname -m) = "aarch64" ] && echo "arm64" || echo "amd64").tar.gz \
+  && wget -q https://dl.google.com/go/$GORELEASE \
+  && tar -C $(dirname $GOROOT) -xzf $GORELEASE \
+  && rm $GORELEASE \
+  && mkdir -p $GOPATH/{src,bin,pkg}
 
 # protoc-gen-go
 ENV PGG_VER=v1.27.1
 RUN go install google.golang.org/protobuf/cmd/protoc-gen-go@${PGG_VER}
 
 # buildozer
-RUN go get github.com/bazelbuild/buildtools/buildozer
+RUN go install github.com/bazelbuild/buildtools/buildozer@latest
```

Comment:
I suspect that this layer will end up being effectively uncacheable, since link-stamping will prevent the buildozer binary from being byte-for-byte identical on rebuilds. Nevertheless, could we try to trim the layer down a bit with 

```
go install google.golang.org/bazelbuild/buildtools/buildozer@latest && rm -rf $(go env GOCACHE) && rm -rf $(go env GOMODCACHE)
```

(`go clean -cache` still leaves timestamped files behind, so I think we really do want a manual `rm -rf` here.)

Reply:
Actually...I think we'll want to make all the `go install` commands possibly do this? I don't actually know much about optimizing builds, but it does seem silly to include the build caches here.

Reply:
resolved in [4a955bb](https://github.com/envoyproxy/protoc-gen-validate/pull/629/commits/4a955bba21cdfcf1b2ecc9a246f35126574cf162)

Reply:
As I understand it, each `RUN` command produces a layer. To get any benefit, we'll have to run the `go install` and the `rm -rf`s in one RUN.